src/procedures/__tests__/TransferErc20.ts

```diff
@@ -0,0 +1,203 @@
+      expect(addTransactionSpy.getCall(0).calledWith(erc20Mock.getMockInstance().transfer)).toEqual(
+        true
+      );
+      expect(addTransactionSpy.getCall(0).lastArg.tag).toEqual(PolyTransactionTag.TransferErc20);
+      expect(addTransactionSpy.callCount).toEqual(1);
+    });
+
+    test('should throw if supplied address does not correspond to a valid erc20 token', async () => {
+      wrappersMock
+        .mock('getERC20TokenWrapper')
+        .withArgs({ address: params.tokenAddress })
+        .throws();
+
+      await expect(target.prepareTransactions()).rejects.toThrow(
+        new PolymathError({
+          code: ErrorCode.ProcedureValidationError,
+          message: 'The supplied address does not correspond to an ERC20 token',
+        })
+      );
+    });
+
+    test('should throw if address belongs to a security token, not an erc20 token', async () => {
+      securityTokenRegistryMock.mock('isSecurityToken', Promise.resolve(true));
+
+      await expect(target.prepareTransactions()).rejects.toThrow(
+        new PolymathError({
+          code: ErrorCode.ProcedureValidationError,
+          message:
+            "This address belongs to a Security Token. To transfer Security Tokens, use the functions in the Security Token's transfers namespace",
+        })
+      );
+    });
+
+    test('should add a transaction to the queue to transfer an erc20 token getting poly token funds if there are not enough funds, only on testnet', async () => {
+      wrappersMock.mock('isTestnet', Promise.resolve(true));
+      polyTokenMock.mock('address', Promise.resolve(params.tokenAddress));
+      erc20Mock.mock('balanceOf', Promise.resolve(new BigNumber(2)));
+      erc20Mock.mock('transfer', Promise.resolve('Transfer'));
+      wrappersMock.mock('getPolyTokens', Promise.resolve('GetPolyTokens'));
+      const addTransactionSpy = spy(target, 'addTransaction');
+      // Real call
+      await target.prepareTransactions();
+
+      // Verifications
+      expect(
+        addTransactionSpy.getCall(0).calledWith(wrappersMock.getMockInstance().getPolyTokens)
+      ).toEqual(true);
+      expect(addTransactionSpy.getCall(0).lastArg.tag).toEqual(PolyTransactionTag.GetTokens);
+      expect(addTransactionSpy.getCall(1).calledWith(erc20Mock.getMockInstance().transfer)).toEqual(
+        true
+      );
+      expect(addTransactionSpy.getCall(1).lastArg.tag).toEqual(PolyTransactionTag.TransferErc20);
+      expect(addTransactionSpy.callCount).toEqual(2);
+    });
+
+    test('should throw error if there are not enough funds to make an erc20 transfer', async () => {
+      erc20Mock.mock('balanceOf', Promise.resolve(new BigNumber(2)));
+
+      await expect(target.prepareTransactions()).rejects.toThrow(
+        new PolymathError({
+          code: ErrorCode.ProcedureValidationError,
+          message: 'Not enough funds',
+        })
+      );
+    });
+
+    test('should successfully create erc20 transfer resolver', async () => {
```

Comment:
Should say something like 'resolver function refreshes the corresponding ERC20 Balance Entity'